packages/@aws-cdk/aws-ec2/lib/peer.ts

```diff
@@ -0,0 +1,166 @@
+import { Connections, IConnectable } from "./connections";
+
+/**
+ * Interface for classes that provide the peer-specification parts of a security group rule
+ */
+export interface IPeer extends IConnectable {
+  /**
+   * Whether the rule can be inlined into a SecurityGroup or not
+   */
+  readonly canInlineRule: boolean;
+
+  /**
+   * A unique identifier for this connection peer
+   */
+  readonly uniqueId: string;
+
+  /**
+   * Produce the ingress rule JSON for the given connection
+   */
+  toIngressRuleJSON(): any;
+
+  /**
+   * Produce the egress rule JSON for the given connection
+   */
+  toEgressRuleJSON(): any;
+}
+
+/**
+ * Factories for static connection peer
+ */
+export class Peer {
+  /**
+   * Create an IPv4 peer from a CIDR
+   */
+  public static cidrIpv4(cidrIp: string): IPeer {
```

Comment:
I think `ipv4` is sufficient in this case (without "cidr"), and then maybe we can do `ipv4Any` instead of `anyIpv4` so they will be grouped together?